semantic-search/pom.xml

```diff
@@ -54,6 +54,23 @@
             <artifactId>json</artifactId>
             <version>20210307</version>
         </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>5.10.0</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
```

Comment:
One thing I realized when working through Java tests was there's a difference between `junit` and `junit.jupiter`. Basically it seems like Junit Jupiter = Junit 5, while the `junit` package is Junit 4. I had to do some looking up, but this explanation is pretty good: https://stackoverflow.com/questions/48853312/java-junit-5-annotations-differences

Basically I think you can drop the `junit` 4.12 dependency entirely and use the `@Test` annotation from the `org.junit.jupiter.api` package, and leverage the assertions from `org.junit.jupiter.api.Assertions.*`. This is at least what we do in the Java client. 

I think ultimately mixing these two dependencies can cause issues if you're using the older `@Test` annotation with the Jupiter interfaces, and it's probably better to just use Jupiter fully.

Reply:
Ah thanks I'll switch to just `junit.jupter` then

Reply:
Done!